### ISSUE TITLE
Testing UI in current navigation (on-device ui) is hard

### ISSUE THREAD

**Issue by matinzd** [storybookjs/react-native#433]
**Is your feature request related to a problem? Please describe.**
It's an improvement request.

**Describe the solution you'd like**
The current storybook makes UI testing hard since are doing it on device and there is not enough space to see the changes when we open different parts. 

**Describe alternatives you've considered**
Maybe it makes more sense to move addons to a bottom sheet that can be dismissed and see the visual changes at the same time.

**Are you able to assist bring the feature to reality?**
Yes, if I find a free time. I already took a look at the code and I am somehow familiar how it works.

**Additional context**

The actual story page size when we expand addons:

<img width="577" alt="Screenshot 2023-02-24 at 23 50 42" src="https://user-images.githubusercontent.com/24797481/221314166-859e9c4c-5c20-4f09-9c68-05513565018f.png">

**Comment by dannyhw** [comment]
@matinzd hey, thanks for sharing your thoughts on this. The short version is yes I think this is the right direction to go, using a bottom sheet would be better and I'm very happy for any PR with an implementation for this.


The long version is that this is something I've had planned for a while (trying out a bottom sheet for addons). And I worked on something for this a while ago but I've been focused on v6 compatibility so i put it aside. 

All this to say that I'm very open to this idea, however there are a few things to consider.

Ideally we shouldn't add more dependencies that aren't pure javascript (unless there is a really good reason). Reanimated would make this really easy to do but that forces everyone who uses storybook to adopt reanimated and I think that's not what we want.

So that leaves us with either implementing it using the rn animated apis or finding a good package that already does this.

I also want to know how this would change the current UI. Opening a bottom sheet from a tab wouldn't make sense really so where would the bottom sheet get triggered from? If addons isn't a tab anymore then does it make sense to use a tab style navigation at all?

Basically when I think about this in more detail it brings up a lot of questions around how it should work.

How would you go about this? How do you think it should work?

**Comment by matinzd** [comment]
I am also really in favor of not adding more native dependencies so that everyone can use the storybook without adding too much dependencies. 

For bottom sheet we have two options. 

- [react-native-modalize](https://github.com/jeremybarbet/react-native-modalize) (uses rn animated api)
- [@gorhom/bottom-sheet](https://github.com/gorhom/react-native-bottom-sheet) (uses reanimated api)

From a UI perspective, I think we can discuss more on that. I have some ideas but they are not organized.

**Comment by jonathanj** [comment]
I've thought about this UI too, although my conclusion was that it is not really a meaningful improvement to the Storybook workflow, maybe I'm missing something.

The way I see it:
- The sheet will completely obscure the story view when it is opened, and once the sheet is dismissed it would need to be opened again to interact with the controls. So in terms of number of taps it remains the same (tap addons, tap close vs tap addons, tap preview)
- In terms of being able to see/interact with your changes it's the same or worse, since you have to move the sheet to see anything at all.

And then as @dannyhw mentioned there needs to be some way to summon the story list too.

**Comment by dannyhw** [comment]
I think for me the main issue with the current tabs implementation is how much the preview shrinks and makes it impossible to see anything. 

Having maybe a way to bring the addons out to a bottom sheet could be better for many components like buttons/textinputs etc which mostly take up horizontal space. 

Actually if you look at the web ui you can move the addons panel to the bottom or the side which is kind of the same thing.


So maybe a button that lets you switch the position of the addons is the better version of this since its also an existing pattern on the web.

**Comment by jonathanj** [comment]
> So maybe a button that lets you switch the position of the addons is the better version of this since its also an existing pattern on the web.

I like this idea, most phones are longer than they are wide anyway.

**Comment by matinzd** [comment]
> > So maybe a button that lets you switch the position of the addons is the better version of this since its also an existing pattern on the web.
> 
> 
> 
> I like this idea, most phones are longer than they are wide anyway.

Exactly. That's the issue we are mostly facing right now. We need to shrink in order to access controls and we can't see component changes in real-time.
Any way we can improve this would be appreciated 🙂

**Comment by dannyhw** [comment]
I believe this is now solved by #438, what do you think @matinzd ?

**Comment by matinzd** [comment]
> I believe this is now solved by #438, what do you think @matinzd ?

Wow! What a great change and quick action on that 😃 

Thanks 🙏🏻 

I will try it as soon as I find time.
